File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int maxproduct=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            int currentproduct=1;
            for(int j=i;j<nums.length;j++){
                currentproduct=currentproduct*nums[j];
                maxproduct=Math.max(maxproduct,currentproduct);
            }
        }
        return maxproduct;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {
    if(nums==null || nums.length==0){
        return 0;
    }
    let maxProduct=nums[0];
    for(let i=0;i<nums.length;i++){
        let currentPro=1;
        for(let j=i;j<nums.length;j++){
            currentPro=currentPro*nums[j];
            if(currentPro>maxProduct){
                maxProduct=currentPro;
            }
        }
    }
    return maxProduct;
};